src/java.naming/share/classes/com/sun/jndi/ldap/Connection.java

```diff
@@ -278,79 +278,86 @@ private InetSocketAddress createInetSocketAddress(String host, int port) {
      * is created.
      */
     private Socket createSocket(String host, int port, String socketFactory,
-            int connectTimeout) throws Exception {
+                                int connectTimeout) throws Exception {
 
         Socket socket = null;
-
-        if (socketFactory != null) {
-
-            // create the factory
-
-            @SuppressWarnings("unchecked")
-            Class<? extends SocketFactory> socketFactoryClass =
-                (Class<? extends SocketFactory>)Obj.helper.loadClass(socketFactory);
-            Method getDefault =
-                socketFactoryClass.getMethod("getDefault", new Class<?>[]{});
-            SocketFactory factory = (SocketFactory) getDefault.invoke(null, new Object[]{});
-
-            // create the socket
-
-            if (connectTimeout > 0) {
-
-                InetSocketAddress endpoint =
-                        createInetSocketAddress(host, port);
-
-                // unconnected socket
-                socket = factory.createSocket();
-
-                if (debug) {
-                    System.err.println("Connection: creating socket with " +
-                            "a timeout using supplied socket factory");
-                }
-
-                // connected socket
-                socket.connect(endpoint, connectTimeout);
+        try {
+            InetSocketAddress endpoint =
+                    createInetSocketAddress(host, port);
+
+            if (socketFactory != null) {
+                // create a connected socket with factory
+                socket = createConnectionSocket(endpoint, socketFactory, connectTimeout);
+            } else { // NO SocketFactory
+                // create a connected socket without factory
+                createConnectionSocket(endpoint, connectTimeout);
```

Comment:
A socket returned by `createConnectionSocket` here needs to be saved to the `socket` variable - currently almost all tests that are not using a custom socket factory - fails with:
```
Caused by: java.lang.NullPointerException: Cannot invoke "java.net.Socket.getInputStream()" because "this.sock" is null
	at java.naming/com.sun.jndi.ldap.Connection.<init>(Connection.java:238)
```